source/Halibut/Transport/ConnectionManagerAsync.cs

```diff
@@ -99,7 +119,7 @@ public async Task<IConnection> AcquireConnectionAsync(ExchangeProtocolBuilder ex
         Tuple<IConnection, Action> CreateNewConnection(ExchangeProtocolBuilder exchangeProtocolBuilder, IConnectionFactory connectionFactory, ServiceEndPoint serviceEndpoint, ILog log, CancellationToken cancellationToken)
         {
             var lazyConnection = new Lazy<IConnection>(() => connectionFactory.EstablishNewConnection(exchangeProtocolBuilder, serviceEndpoint, log, cancellationToken));
-            var connection = new DisposableNotifierConnection(lazyConnection, OnConnectionDisposed);
+            var connection = new DisposableNotifierConnection(lazyConnection, OnConnectionDisposed, OnConnectionDisposedAsync);
```

Comment:
Note: This sync implementation is slated for removal for the context of other readers.